test/integration/image-component/basic/pages/missing-observer.js

```diff
@@ -1,12 +1,14 @@
-import React, { useLayoutEffect } from 'react'
 import Image from 'next/image'
+import Script from 'next/script'
 
 const Lazy = () => {
-  useLayoutEffect(() => {
-    IntersectionObserver = null //eslint-disable-line
-  })
   return (
     <div>
+      <Script
+        id="IntersectionObserver-null"
+        strategy="beforeInteractive"
+        dangerouslySetInnerHTML={{ __html: 'IntersectionObserver = null' }}
+      />
```

Comment:
`hasIntersectionObserver` is defined when the module is loaded. So we shouldn't dynamically change the value during effect.

Reply:
re https://github.com/vercel/next.js/pull/37447#discussion_r890303552

cc @styfle